source/particles/utilities.cc

```diff
@@ -0,0 +1,221 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/lac/generic_linear_algebra.h>
+
+#include <deal.II/particles/utilities.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace Particles
+{
+  namespace Utilities
+  {
+    template <int dim, int spacedim, typename Sparsity, typename number>
+    void
+    create_interpolation_sparsity_pattern(
+      const DoFHandler<dim, spacedim> &                space_dh,
+      const Particles::ParticleHandler<dim, spacedim> &particle_handler,
+      Sparsity &                                       sparsity,
+      const AffineConstraints<number> &                constraints,
+      const ComponentMask &                            space_comps)
+    {
+      if (particle_handler.n_locally_owned_particles() == 0)
+        return; // nothing to do here
+
+      const auto &tria     = space_dh.get_triangulation();
+      const auto &fe       = space_dh.get_fe();
+      auto        particle = particle_handler.begin();
```

Comment:
Move this line just above the first place where you use it, i.e., the `while` loop.